src/session.ts

```diff
@@ -165,19 +166,23 @@ export class SessionManager implements Middleware {
                 }
             }
 
+            // Generate a random id for the named pipes in case they have multiple instances of PSES running
+            const id = crypto.randomBytes(10).toString("hex");
             this.editorServicesArgs =
-                "-HostName 'Visual Studio Code Host' " +
-                "-HostProfileId 'Microsoft.VSCode' " +
-                "-HostVersion '" + this.hostVersion + "' " +
-                "-AdditionalModules @('PowerShellEditorServices.VSCode') " +
-                "-BundledModulesPath '" + this.bundledModulesPath + "' " +
-                "-EnableConsoleRepl ";
+                `-HostName 'Visual Studio Code Host' ` +
+                `-HostProfileId 'Microsoft.VSCode' ` +
+                `-HostVersion '${this.hostVersion}'` +
+                `-AdditionalModules @('PowerShellEditorServices.VSCode') ` +
+                `-BundledModulesPath '${this.bundledModulesPath}'` +
+                `-EnableConsoleRepl ` +
+                `-LanguageServicePipeName LanguageService_${id}.pipe ` +
+                `-DebugServicePipeName DebugService_${id}.pipe `;
 
             if (this.sessionSettings.developer.editorServicesWaitForDebugger) {
-                this.editorServicesArgs += "-WaitForDebugger ";
+                this.editorServicesArgs += `-WaitForDebugger `;
             }
             if (this.sessionSettings.developer.editorServicesLogLevel) {
-                this.editorServicesArgs += "-LogLevel '" + this.sessionSettings.developer.editorServicesLogLevel + "' ";
+                this.editorServicesArgs += `-LogLevel '${this.sessionSettings.developer.editorServicesLogLevel}' `;
```

Comment:
+1

Reply:
why don't we need it here?

Reply:
If your question is about the string interpolation, then I'm for its use here, where there is a variable to interpolate inside the string.  :-)